### PR TITLE
Adding argument to specify version of syw to install

### PR DESCRIPTION
**Pull request by dfm**
The syntax is a little ugly and open to discussion, but this allows syntax like:

```yaml
      - name: Build the article PDF
        id: build
        uses: showyourwork/showyourwork-action@v1
        with:
          pip-install-syw: git+https://github.com/dfm/showyourwork.git#egg=showyourwork
```

to install syw from a fork. I expect this would be a pretty advanced use case so I'm not too sad if the syntax isn't great, but it could be better!

See updated syntax below.

### CODE REVIEW

**Comment by dfm** [comment]
I'm using this over in https://github.com/dfm/araa-gps and it's working. I'm going to play around with some alternate syntax choices.

File: src/conda.js [hunk 1]
```diff
@@ -51,7 +51,10 @@ async function setupConda() {
   }
 
   // Always install the latest version of showyourwork
-  exec("pip install -U showyourwork", "Install showyourwork");
+  const version = core.getInput("showyourwork-version");
```
Comment (rodluger) [inline src/conda.js:54]: Doesn't look like the `showyourwork-version` option has any effect on the code.

**Review by rodluger — CHANGES_REQUESTED** [review]

File: src/conda.js (hunk 1 above)
Comment (dfm) [reply src/conda.js:54]: That's correct - I am experimenting :D The actual parameter is currently `pip-install-syw`.

Comment (rodluger) [reply src/conda.js:54]: Got it. Carry on!

**Comment by dfm** [comment]
OK I think this is better now. The syntax is somewhat similar to the one in https://github.com/showyourwork/showyourwork/pull/148. You can write things like:

```yaml
      - uses: showyourwork/showyourwork-action@v1
        with:
          showyourwork-fork: https://github.com/dfm/showyourwork.git
          showyourwork-ref: branch-name
```

or

```yaml
      - uses: showyourwork/showyourwork-action@v1
        with:
          showyourwork-version: 0.3.0
```

or

```yaml
      - uses: showyourwork/showyourwork-action@v1
        with:
          showyourwork-ref: main
```

**Comment by dfm** [comment]
I found an issue with this. I'll be back to is soon.

**Comment by rodluger** [comment]
Deprecating this since I already implemented (and deployed) something similar:

https://github.com/showyourwork/showyourwork-action/blob/116e0ef5cf649359d2a12fab54e1cc0f967e517c/action.yml#L45-L48

https://github.com/showyourwork/showyourwork-action/blob/116e0ef5cf649359d2a12fab54e1cc0f967e517c/src/conda.js#L55

No need for the granularity at the input level (version, fork, ref) -- easier if the input is simply the spec passed to `pip`.

**Pull request closed without merging**